### PR TITLE
Make Rust main code in the repository

### PR DESCRIPTION
**Pull request by abailly-iohk**
This inverts the relationship between Rust and Haskell in this repo: The Rust code is the primary code, and Haskell code moves to a subdirectory.

### CODE REVIEW

File: haskell/src/ALBA.hs
Comment (djetchev) [inline haskell/src/ALBA.hs:1]: @abailly-iohk - is this file going to be deprecated eventually?

File: caledonia/.gitignore
Comment (djetchev) [inline caledonia/.gitignore:1]: The `caledonia/` directory gets deprecated.

File: haskell/src/ALBA.hs
Comment (abailly-iohk) [reply haskell/src/ALBA.hs:1]: Yes

File: caledonia/.gitignore
Comment (abailly-iohk) [reply caledonia/.gitignore:1]: good catch

Comment (abailly-iohk) [reply caledonia/.gitignore:1]: hmm actually yes, its' been removed

**Pull request merged**